de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/shipmentschedule/api/ShipmentScheduleWithHU.java

```diff
@@ -129,9 +129,14 @@ public static final ShipmentScheduleWithHU ofShipmentScheduleWithoutHu(
 
 	private I_M_InOutLine shipmentLine = null;
 
+
+	private M_ShipmentSchedule_QuantityTypeToUse qtyTypeToUse = M_ShipmentSchedule_QuantityTypeToUse.TYPE_D; // keep old functionality as default
```

Comment:
qtyTypeToUse and isForPickedQty  seem to be sortof redundant;
doesn't TYPE_D and TYPE_BD imply that isForPickedQty = true ?

In that case, you could kick out `isForPickedQty ` and write a method `isForPickedQty()` that uses qtyTypeToUse

Reply:
when looking at the places where `setQtyTypeToUse()` is called, i think it makes sense to 
extend on `ShipmentScheduleWithHU.ofShipmentSchedule...()` instead of having the public setter

Reply:
Done.
I had to keep also the other version, without QtyTypeToUse, because these methods are not only used by the shipment creator. In some places we don't have this info, e.g. when we link QtyPicked with shipment schedules.